controllers/user.controller.js

```diff
@@ -35,28 +66,41 @@ export const updateUser = async (req, res, next) => {
       return res.status(404).json({ message: 'User not found.' });
     }
 
+    let hashedPassword = userExistsResult.rows[0].password;
+    if (password) {
+      const saltRounds = 10;
+      hashedPassword = await bcrypt.hash(password, saltRounds);
+    }
+
     const updateUserQuery = `
       UPDATE users
-      SET username = $1, age = $2, email = $3
-      WHERE id = $4
+      SET username = $1, age = $2, email = $3, role = $4, password = $5
+      WHERE id = $6
       RETURNING *;
     `;
-    const updateUserValues = [username, age, email, id];
+
+    const updateUserValues = [username, age, email, role, hashedPassword, id];
```

Comment:
Why are we retrieving the password on an account update? I don't think that's necessary. Also, you do not want to send `password` in a request body, after account creation. That's what JWTs are for. Passwords aren't needed to verify the user after the user has already signed in